armi/reactor/converters/tests/test_axialExpansionChanger.py

```diff
@@ -1266,3 +1273,109 @@ def linearExpansionPercent(self, Tk=None, Tc=None):
         """A fake linear expansion percent."""
         Tc = units.getTc(Tc, Tk)
         return 0.08 * Tc
+
+
+class TestAxialLinkHelper(unittest.TestCase):
+    """Tests for the AxialLink dataclass / namedtuple like class."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.LOWER_BLOCK = _buildDummySodium(20, 10)
+        cls.UPPER_BLOCK = _buildDummySodium(300, 50)
+
+    def setUp(self):
+        self.link = AxialLink(self.LOWER_BLOCK, self.UPPER_BLOCK)
+
+    def test_positionGet(self):
+        """Test the getitem accessor."""
+        self.assertIs(self.link[0], self.link.lower)
+        self.assertIs(self.link[1], self.link.upper)
+        with self.assertRaises(AttributeError):
+            self.link[3]
+
+    def test_positionSet(self):
+        """Test the setitem setter."""
+        self.link[0] = None
+        self.assertIsNone(self.link.lower)
+        self.link[0] = self.LOWER_BLOCK
+        self.assertIs(self.link.lower, self.LOWER_BLOCK)
+        self.link[1] = None
+        self.assertIsNone(self.link.upper)
+        self.link[1] = self.UPPER_BLOCK
+        self.assertIs(self.link.upper, self.UPPER_BLOCK)
+        with self.assertRaises(AttributeError):
+            self.link[-1] = None
+
+    def test_iteration(self):
+        """Test the ability to iterate over the items in the link."""
```

Comment:
just curious, why do we need to have the ability to iterate here? We only ever access AxialLink.lower or AxialLink.upper, right?

Reply:
Back compat? I think it was needed in earlier versions but let me see about removing `AxialLink.__iter__`

Reply:
https://github.com/terrapower/armi/pull/1920/commits/1cc156a9cd0e6bf0c8c6fa33003461bd48636911